src/main/java/org/jfrog/hudson/PipelinesServer.java

```diff
@@ -0,0 +1,217 @@
+                // 'JFrogPipelines' parameter is not set.
+                return;
+            }
+            if (!(isConfigured(pipelinesServer))) {
+                // JFrog Pipelines server is not configured, but 'JFrogPipelines' parameter is set.
+                logger.error(SERVER_NOT_FOUND_EXCEPTION);
+                return;
+            }
+            String stepId = jfrogPipelinesParam.getStepId();
+            if (pipelinesServer.isReported(stepId)) {
+                // Step status is already reported to JFrog Pipelines.
+                pipelinesServer.clearReported(stepId);
+                logger.debug("Skipping reporting to JFrog Pipelines - status is already reported in jfPipelines step.");
+                return;
+            }
+            pipelinesServer.reportNow(build.getResult(), jfrogPipelinesParam.getStepId(), logger);
+        } catch (IOException | IllegalArgumentException | IllegalStateException e) {
+            if (isConfigured(pipelinesServer)) {
+                // If JFrog Pipelines server is not configured don't log errors.
+                // This case is feasible when build.getEnvironment throws an exception.
+                logger.error(FAILURE_PREFIX + ExceptionUtils.getRootCauseMessage(e), e);
+            }
+        }
+    }
+
+    /**
+     * Send information to JFrog Pipelines after a pipeline job finished or when a reportNow step invoked.
+     * Input parameter:
+     * { stepId: <JFrog Pipelines step ID> }
+     * Output:
+     * {
+     * action: "status",
+     * status: <Jenkins build status>,
+     * stepId: <JFrog Pipelines step ID>
+     * }
+     *
+     * @param result - The build results
+     * @param stepId - JFrog Pipelines step ID
+     * @param logger - The build logger
+     */
+    public void reportNow(Result result, String stepId, JenkinsBuildInfoLog logger) throws IOException {
+        try (PipelinesHttpClient client = createPipelinesHttpClient(credentialsConfig.provideCredentials(null), ProxyUtils.createProxyConfiguration(), logger)) {
+            client.jobCompleted(new JobCompletedPayload(result, stepId, getOutputResource(stepId)));
+        }
+        logger.info("Successfully reported status '" + result + "' to JFrog Pipelines.");
+    }
+
+    /**
+     * Get JFrog Pipelines server from the global configuration.
+     *
+     * @return configured JFrog Pipelines server
+     * @throws IllegalStateException if Pipelines server is not defined
+     */
+    public static PipelinesServer getPipelinesServer() {
+        ArtifactoryBuilder.DescriptorImpl descriptor =
+                (ArtifactoryBuilder.DescriptorImpl) Jenkins.get().getDescriptor(ArtifactoryBuilder.class);
+        if (descriptor == null) {
+            return null;
+        }
+        PipelinesServer pipelinesServer = descriptor.getPipelinesServer();
+        if (StringUtils.isBlank(pipelinesServer.getCbkUrl())) {
+            return null;
+        }
+        return pipelinesServer;
```

Comment:
Null is a a possible value. If the user hasn't defined JFrog Pipelines server we may want to avoid reporting the jobs' statuses to JFrog Pipelines.
`isConfigured` is checking if the return value is null or empty.
I fixed the Javadoc accordingly.
Please let me know if you agree.